modules/core/src/test/java/org/apache/ignite/lang/GridXSelfTest.java

```diff
@@ -53,6 +54,42 @@ public void testHasCause() {
         assert gridEx.getCause(NumberFormatException.class) == null;
     }
 
+    /**
+     * Checks that using function with Throwable that contains circular reference in {@code getSuppressed()}
+     *     does not cause {@link StackOverflowError}.
+     */
+    @Test
+    public void testCauseDoesNotRaiseStackOverflowWhenCircularReferenceInSuppressed() {
```

Comment:
Add a test that did not find an exception

Reply:
added asserts to existing tests for this case.